internal/util/util.go

```diff
@@ -474,9 +474,12 @@ func EscapePathLegacy(path string) string {
 // escaped using URL encoding or legacy encoding with '%' (for backward
 // compatibility, if the escaped path exists in the given directory).
 func DeterminePath(dir string, path string) string {
-	md5sum := filepath.Join(dir, HashStringMd5(path))
-	if _, err := os.Stat(md5sum); err == nil {
-		return md5sum
+	runes := []rune(filepath.Base(path))
+	truncBaseName := string(runes[len(runes)-Min(len(runes), 16):])
```

Comment:
Why the last 16 characters, not the first 16 characters?

Reply:
I suggested last N characters so the file extension is kept, but in this implementation the md5 hash is *after* the truncated name. I would probably swap the parts (`<md5>_<last16>` instead of `<last16>_<md5>`).